File: src/assets/usage-examples-code/graphics/fill_triangle_on_bitmap/fill_triangle_on_bitmap-1-basic-bitmap.cpp
#include "splashkit.h"

int main()
{
    // Open a window
    open_window("Happy Hat", 618, 618);

    // Load the bitmaps for sad and smiling emojis (https://openmoji.org/library/#group=smileys-emotion)
    bitmap sad_emoji = load_bitmap("sad_emoji", "sad_emoji.png");
    bitmap smiling_emoji = load_bitmap("smiling_emoji", "smiling_emoji.png");

    // Draw the sad emoji and add a hat
    clear_screen(COLOR_BLACK);
    draw_bitmap(sad_emoji, 0, 0);
    refresh_screen();
    delay(1000);

    // Draw a triangle hat on the smiling emoji
    fill_triangle_on_bitmap(smiling_emoji, COLOR_RED, 100, 200, 309, 20, 520, 200);

    // Clear screen and switch to the smiling emoji
    clear_screen(COLOR_BLACK);
    draw_bitmap(smiling_emoji, 0, 0);
    refresh_screen();
    delay(1000);

    // Spin the smiling emoji with the hat
    for (int i = 0; i < 360; i++)
    {
        clear_screen(COLOR_BLACK);
        draw_bitmap(smiling_emoji, 0, 0, option_rotate_bmp(i));
        refresh_screen();
        delay(10);
    }

    // Free the bitmap resource
    free_all_bitmaps();
    // Close all windows
    close_all_windows();

    return 0;
}
 

File: src/assets/usage-examples-code/graphics/fill_triangle_on_bitmap/fill_triangle_on_bitmap-1-basic-bitmap.cs
using SplashKitSDK;
using static SplashKitSDK.SplashKit;

OpenWindow("Underwater Adventure", 1000, 500);

// Load the bitmap
Bitmap underwaterScene = LoadBitmap("underwater_scene", "underwater_scene.png");

// Fill the triangle on the bitmap
double x1 = 285, y1 = 250;
double x2 = 310, y2 = 240;
double x3 = 310, y3 = 260;

FillTriangleOnBitmap(underwaterScene, ColorRed(), x1, y1, x2, y2, x3, y3);

// Draw the bitmap and other graphics
DrawBitmap(underwaterScene, 0, 0);
DrawText("Colourful Fish", ColorRed(), "arial", 123, 240, 200);
DrawLine(ColorRed(), 280, 220, 305, 255);

// Refresh the screen and wait
RefreshScreen();
Delay(5000);

// Close all windows
CloseAllWindows();
